src/Plugins/ComplexCore/src/ComplexCore/Filters/ImportCSVDataFilter.cpp

```diff
@@ -0,0 +1,487 @@
+  }
+
+  return {};
+}
+
+// -----------------------------------------------------------------------------
+Result<OutputActions> validateExistingGroup(const DataPath& groupPath, const DataStructure& dataStructure, const std::vector<std::string>& headers)
+{
+  if(groupPath.empty())
+  {
+    return {MakeErrorResult<OutputActions>(IssueCodes::EMPTY_EXISTING_DG, "'Existing Data Group' - Data path is empty.")};
+  }
+
+  const BaseGroup& selectedGroup = dataStructure.getDataRefAs<BaseGroup>(groupPath);
+  const auto arrays = selectedGroup.findAllChildrenOfType<IDataArray>();
+  for(const std::shared_ptr<IDataArray>& array : arrays)
+  {
+    std::string arrayName = array->getName();
+    for(const std::string& headerName : headers)
+    {
+      if(arrayName == headerName)
+      {
+        return {MakeErrorResult<OutputActions>(IssueCodes::DUPLICATE_NAMES, fmt::format("The header name \"{}\" matches an array name that already exists in the selected container.", headerName))};
+      }
+      if(StringUtilities::contains(headerName, '&') || StringUtilities::contains(headerName, ':') || StringUtilities::contains(headerName, '/') || StringUtilities::contains(headerName, '\\'))
+      {
+        return {MakeErrorResult<OutputActions>(IssueCodes::ILLEGAL_NAMES,
+                                               fmt::format("The header name \"{}\" contains a character that will cause problems. Do Not use '&',':', '/' or '\\' in the header names.", headerName))};
+      }
+    }
+  }
+
+  return {};
+}
+
+// -----------------------------------------------------------------------------
+Result<OutputActions> validateNewGroup(const DataPath& groupPath, const DataStructure& dataStructure)
+{
+  if(groupPath.empty())
+  {
+    return {MakeErrorResult<OutputActions>(IssueCodes::EMPTY_NEW_DG, "'New Data Group' - Data path is empty.")};
+  }
+
+  if(dataStructure.getData(groupPath) != nullptr)
+  {
+    return {MakeErrorResult<OutputActions>(IssueCodes::NEW_DG_EXISTS, fmt::format("The group at the path '{}' cannot be created because it already exists.", groupPath.toString()))};
+  }
+
+  return {};
+}
+
+// -----------------------------------------------------------------------------
+Result<ParsersVector> createParsers(const DataTypeVector& dataTypes, const DataPath& parentPath, const std::vector<std::string>& headers, DataStructure& dataStructure)
+{
+  ParsersVector dataParsers(dataTypes.size());
+
+  for(usize i = 0; i < dataTypes.size(); i++)
+  {
+    std::optional<DataType> dataTypeOpt = dataTypes[i];
+    if(!dataTypeOpt.has_value())
+    {
+      continue;
+    }
+
+    std::string name = headers[i];
+
+    DataPath arrayPath = parentPath;
+    arrayPath = arrayPath.createChildPath(name);
+
+    DataType dataType = dataTypeOpt.value();
+    if(dataType == complex::DataType::float64)
```

Comment:
```suggestion
    switch(dataType)
```
This is already structured like a switch statement and checks against an enum value. Might as well use one.